test-framework/junit5-component/src/main/java/io/quarkus/test/component/QuarkusComponentTestExtension.java

```diff
@@ -1200,10 +1201,10 @@ private File getTestOutputDirectory(Class<?> testClass) {
             // org.acme.Foo -> org/acme/Foo.class
             String testClassResourceName = fromClassNameToResourceName(testClass.getName());
             // org/acme/Foo.class -> /some/path/to/project/target/test-classes/org/acme/Foo.class
-            String testPath = testClass.getClassLoader().getResource(testClassResourceName).getFile();
+            String testPath = testClass.getClassLoader().getResource(testClassResourceName).toString();
```

Comment:
I don't like this because it doesn't do what the comment says. If fact, the original code also doesn't do what the comment says.

The `getResource` method returns a `java.net.URL`. A `URL` always has a protocol (relative URLs are not allowed). So, `getFile` turns e.g. `file:/foo%20bar/Foo.class` into `/foo%20bar/Foo.class`. Misleading. But `toString()` turns e.g. `file:/foo%20bar/Foo.class` into `file:/foo%20bar/Foo.class` (but as a string). Also bad, and not even close to the comment.

Down below, we next trim off the class part, which is fine, giving us `file:/foo%20bar/` (note we have a trailing slash there, I think the math is off by one because the path of the class does not include its leading `/`).

Finally we create a new URI off of the trimmed URL string, giving a URI of `file:/foo%20bar`. Since `java.net.URI` has slightly more intelligent string processing, the `getPath` operation which is implicitly performed by `new File(uri)` gives `foo bar`. Hooray! But confusing... how did we get here?

It would be good to make sure the comments clearly express what's going on here. `URI` and `URL` are frankly really bad and confusing classes. I always end up going to `jshell` to find out what *really* happens when you give a certain kind of input, and I'm often surprised by the result.

Reply:
I found a much easier way to do the exact same thing, without any manual path manipulation. See the new push.

In the old setup, we would construct the on-disk path to `testClass` using `fromClassNameToResourceName`. We would then use this path to find a resource (relative to the test classes root), and then spend time to "undo" this navigation to get back to the test classes root. The new approach just finds the path to the test classes root directly using `"."`. I tested this in another project as well, and also there the path was correct and as expected.